pkg/operator/target_config_reconciler.go

```diff
@@ -0,0 +1,364 @@
+// compatibility with 3.11
+var SecondarySchedulerCommand = []string{"/bin/kube-scheduler", "--config", "/etc/kubernetes/config.yaml", "--v", "2"}
+
+type TargetConfigReconciler struct {
+	ctx                      context.Context
+	targetImagePullSpec      string
+	operatorClient           operatorconfigclientv1.SecondaryschedulersV1Interface
+	secondarySchedulerClient *operatorclient.SecondarySchedulerClient
+	kubeClient               kubernetes.Interface
+	osrClient                openshiftrouteclientset.Interface
+	dynamicClient            dynamic.Interface
+	eventRecorder            events.Recorder
+	queue                    workqueue.RateLimitingInterface
+	excludedNamespaces       []string
+}
+
+func NewTargetConfigReconciler(
+	ctx context.Context,
+	targetImagePullSpec string,
+	operatorConfigClient operatorconfigclientv1.SecondaryschedulersV1Interface,
+	operatorClientInformer operatorclientinformers.SecondarySchedulerInformer,
+	secondarySchedulerClient *operatorclient.SecondarySchedulerClient,
+	kubeClient kubernetes.Interface,
+	osrClient openshiftrouteclientset.Interface,
+	dynamicClient dynamic.Interface,
+	eventRecorder events.Recorder,
+) *TargetConfigReconciler {
+	// make sure our list of excluded system namespaces is up to date
+	allNamespaces, err := kubeClient.CoreV1().Namespaces().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		klog.ErrorS(err, "error listing namespaces")
+		return nil
+	}
+	excludedNamespaces := []string{"kube-system"}
+	for _, ns := range allNamespaces.Items {
+		if strings.HasPrefix(ns.Name, "openshift-") {
+			excludedNamespaces = append(excludedNamespaces, ns.Name)
+		}
+	}
+
+	c := &TargetConfigReconciler{
+		ctx:                      ctx,
+		operatorClient:           operatorConfigClient,
+		secondarySchedulerClient: secondarySchedulerClient,
+		kubeClient:               kubeClient,
+		osrClient:                osrClient,
+		dynamicClient:            dynamicClient,
+		eventRecorder:            eventRecorder,
+		queue:                    workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "TargetConfigReconciler"),
+		excludedNamespaces:       excludedNamespaces,
+		targetImagePullSpec:      targetImagePullSpec,
+	}
+	operatorClientInformer.Informer().AddEventHandler(c.eventHandler())
+
+	return c
+}
+
+func (c TargetConfigReconciler) sync() error {
+	secondaryScheduler, err := c.operatorClient.SecondarySchedulers(operatorclient.OperatorNamespace).Get(c.ctx, operatorclient.OperatorConfigName, metav1.GetOptions{})
+	if err != nil {
+		klog.ErrorS(err, "unable to get operator configuration", "namespace", operatorclient.OperatorNamespace, "secondary-scheduler", operatorclient.OperatorConfigName)
+		return err
+	}
+
+	forceDeployment := false
+	_, forceDeployment, err = c.manageConfigMap(secondaryScheduler)
+	if err != nil {
+		return err
+	}
+
+	// if _, _, err := c.manageServiceAccount(secondaryScheduler); err != nil {
+	// 	return err
+	// }
```

Comment:
why is this commented out?

Reply:
Yeah, I will uncomment this. I was testing creating the service account via the deploy folder and see if we can only create one service account for both secondary-scheduler and secondary-scheduler-operator just like the example in descheduler case.